lib/ukrandom/chacha.c

```diff
@@ -156,13 +157,28 @@ int uk_swrand_init(void)
 	int ret;
 
 	uk_pr_info("Initialize random number generator...\n");
-
+#if CONFIG_LIBUKRANDOM_SEED_INSECURE
+	uk_pr_err("*******************************************\n");
```

Comment:
Why not `uk_pr_warn`?

Reply:
Because we want these to be printed even if the severity has been set to low. And also to be printed in red if `LIBUKDEBUG_ANSI_COLOR` is set.

Reply:
Will the catalog have these shown? I feel like this is still too agressive and misleading as it is not an error.

> we want these to be printed even if the severity has been set to low

Why not `crit` then?

Reply:
It should be aggressive, if you disable the CSRNG the security of the entire system is compromised. I am not using `crit` because according to @skuenzer that level should only used when we crash Unikraft.

Reply:
Still, error feels wrong as it is not an error and you can still configure Unikraft to not show errors the same way you can configure it to not show warnings :thinking:.

Reply:
I agree, normally we would have used `warn` for this however I think this particular case is a very special and requires increasing the seriousness of the message with `err`.
The purpose of `lib/ukrand` is to provide cryptographically secure pseudo random numbers. Anything just random without crypotgraphically security can and should be done with the libc (nolibc has the according `randr()` call as well). We introduce only one option to make a system deterministic for debugging purposes (and only for this use case) via a compiled-in seed. For example, this can be useful when we continue implementing ASLR. Any other cryptographically insecure random number generators are not allowed in this lib because it breaks assumptions that application code relies on. So, the error is there to make this dominant, and yes, it is maybe arguable if breaking the assumption is a warning or an error but because we want to provide a secure environment with the Unikraft project and avoid configuration mistake as much as possible (e.g., by mistake an image configured insecurely) we should increase the seriousness. When configuring images, it feels easier to hide warnings, but not errors. And since we consider broken CSRNG to be a serious flaw, it is an error that should not be overlooked. Of course you always can workaround this (e.g., hiding errors), but the measures we implement make it harder to ignore.

Reply:
Forgot to mention: The "crit" message level is intended for messages that are so serious that the system cannot continue operating, like running into a crash.

Reply:
k